aws/sqs/sqs.go

```diff
@@ -134,7 +137,12 @@ func (s *SQS) receiveMessage(ctx context.Context, input *sqs.ReceiveMessageInput
 
 		switch {
 		case r == nil || len(r.Messages) == 0:
-			// no message received
+			// No message received
+			retryCount++
```

Comment:
Retrying this doesn't seem to help.

No matter it's retried or not,  the caller has to dealt with `IsNoMessageError`, and continue the loop with the same `sqsClient.receiveMewssage()` call.

No matter we have retry here or not, the caller's code doesn't change. And retrying here doesn't save CPU, nor number of API requests. So I suggest that we remove this retry, and let the caller decide whether it wants to count the number of retries.

Reply:
> Retrying this doesn't seem to help.
> 
Retrying just lets it to wait longer for messages to come, and with less interaction

Reply:
Say, the message arrives after 99 aws.receiveMessage.
- The existing implementation takes one call to kit.receive, and no need to deal with empty retires.
   I would say this is the reason we want a retry inside kit.   
- Removing the retry completely takes 99 calls to kit.receive, and in main we need to deal with empty retries.
- Adding 3 retries takes 33 calls to kit.receive, and in main we need to deal with empty retries.

Reducing the number of call to kit.receive doesn't save anything, but adding the complexity of retrying inside kit.